lib/nuget-parser/nuspec-parser.ts

```diff
@@ -8,80 +8,126 @@ const debug = debugModule('snyk');
 
 const targetFrameworkRegex = /([.a-zA-Z]+)([.0-9]+)/;
 
-export async function parseNuspec(dep, targetFramework) {
-  return Promise.resolve()
-    .then(() => {
-      const nupkgPath = path.resolve(
-        dep.path,
-        dep.name + '.' + dep.version + '.nupkg',
+async function loadNuspecFromAsync(dep) {
+  const nupkgPath = path.resolve(
+    dep.path,
+    dep.name + '.' + dep.version + '.nupkg',
+  );
+
+  //just in case, does the code expect null to be returned on any error here?
+  if (!fs.existsSync(nupkgPath)) return null;
```

Comment:
1. Use `await fs.promises.exists(nupkgPath)` instead of the sync call (because it's in use in dotnet-deps and not only on the cli)
2. Please add a debug message about it (as you did in other places)